### PR TITLE
Handle actions on LinkTo with a component

### PR DESCRIPTION
**Pull request by jrjohnson**
This builds an internal LinkToWithAction component that does what we
need in this case without overriding the internal component. It isn't a
great replacement, but it handles this case and removes the deprecation
and linting errors that our original solution has caused.

### CODE REVIEW

File: app/components/user-menu.hbs [hunk 1]
```diff
@@ -24,22 +24,23 @@
...
-          </LinkTo>
+          </LinkToWithAction>
         </li>
         <li tabindex="-1" data-test-item>
-          <LinkTo
+          <LinkToWithAction
             @route="mymaterials"
-            @invokeAction={{set this.isOpen false}}
+            @action={{set this.isOpen false}}
+            @queryParams={{hash course=null filter=null sortBy=null firstOfferingDate=null}}
```
Comment (stopfstedt) [inline app/components/user-menu.hbs:35]: this blows up when you set `sortBy` to `null` with a `TypeError: sortKey is null` error.

```suggestion
            @queryParams={{hash course=null filter=null firstOfferingDate=null}}
```

**Review by stopfstedt — CHANGES_REQUESTED** [review]

File: app/components/user-menu.hbs (hunk 1 above)
Comment (jrjohnson) [reply app/components/user-menu.hbs:35]: This doesn't end up setting the default correctly as it populates it on the URL string. I'm gonna try changing this controller a bit in common to handle this. Good catch!

**Comment by jrjohnson** [comment]
Requires https://github.com/ilios/common/pull/2350

**Review by stopfstedt — APPROVED** [review]
retested with ilios/common#2350, looks good to me.

**Comment by stopfstedt** [comment]
flagged as `DO NOT MERGE` until the next common release has been cut and integrated.

**Pull request merged**